firmware/application/external/rook/ui_rook.cpp

```diff
@@ -0,0 +1,269 @@
+        size_t length = payload.length();
+ 
+        // Ensure waveform length does not exceed buffer size
+        if (length + (PADDING_LEFT + PADDING_RIGHT) >= WAVEFORM_BUFFER_SIZE) {
+            length = WAVEFORM_BUFFER_SIZE - (PADDING_LEFT + PADDING_RIGHT);
+        }
+ 
+        // Left padding
+        for (size_t i = 0; i < PADDING_LEFT; i++) {
+            waveform_buffer[i] = 0;
+        }
+ 
+        // Draw the actual waveform
+        for (size_t n = 0; n < length; n++) {
+            waveform_buffer[n + PADDING_LEFT] = (payload[n] == '0') ? 0 : 1;
+        }
+ 
+        // Right padding
+        for (size_t i = length + PADDING_LEFT; i < WAVEFORM_BUFFER_SIZE; i++) {
+            waveform_buffer[i] = 0;
+        }
+ 
+        waveform.set_length(length + PADDING_LEFT + PADDING_RIGHT);
+        waveform.set_dirty();
+    }
+ 
+    // Destructor for `ROOKAppView`: Disables the transmitter and shuts down the baseband
+    ROOKAppView::~ROOKAppView() {
+        transmitter_model.disable();
+        baseband::shutdown();
+    }
+ 
+    // Constructor for `ROOKAppView`: Sets up the app view and initializes UI elements
+    ROOKAppView::ROOKAppView(NavigationView &nav)
+    : nav_{nav} {
+        baseband::run_image(portapack::spi_flash::image_tag_ook);
+      
+        add_children({
+            &field_frequency,
+            &tx_view,
+            &button_start,
+            &sample_rate,
+            &step_symbol_rate,
+            &cant_step_symbol_rate,
+            &field_sample_rate,
+            &field_symbol_rate,
+            &field_symbol_us_rate,
+            &cant_symbol_rate,
+            &text_payload,
+            &button_set,
+            &progressBar_progress,
+            &repeat,
+            &cant_repeat,
+            &field_pause_symbol,
+            &cant_pause_symbol,
+            &label_waveform,
+            &waveform,
+            &button_open,
+        });
+ 
+        // Initialize default values for controls
+        cant_pause_symbol.set_value(100);
+        cant_repeat.set_value(4);
+ 
+        button_open.on_select = [this](Button&) {
+            auto open_view = nav_.push<FileLoadView>(".TXT");
+            open_view->on_changed = [this](std::filesystem::path new_file_path) {
+                // Postpone `on_file_changed` call until `FileLoadView` is closed
+                nav_.set_on_pop([this, new_file_path]() {
+                    on_file_changed(new_file_path);
+                    button_start.focus();
+                    draw_waveform();
+                });
+            };
+        };
+ 
+        // Set up changes for symbol rate and step
+        cant_symbol_rate.on_change = [this](int32_t value) {
+            if (value != cant_symbol_rate.value())
+                cant_symbol_rate.set_value(value);
+        };
+ 
+        cant_step_symbol_rate.on_change = [this](size_t, int32_t value) {
+            cant_symbol_rate.set_step(value);
```

Comment:
If the step is only for symbol rate, it would be nice to either
- move it's position off to frequency widgets to prevent misleading
- rename "Step" label into something like "Symbol Field Step" (maybe not enough screen space tho, so maybe still need to edit the position)

These are just suggestions. you are free to choose accept or not

Reply:
"SymRate" looks ok, no ?

Reply:
Made adjustments to widget positioning to replace Step: with Step/symbols